src/parsers/ddi_parser.jl

```diff
@@ -0,0 +1,174 @@
+
+
+"""
+    parse_ddi(filepath::String)
+
+Parses a valid IPUMs DDI XML file and returns a DDIInfo object containing 
+the IPUMs extract metadata. 
+
+### Arguments
+
+- `filepath::String` -- A string containing the path to the IPUMS DDI XML file. 
+
+### Returns
+
+A DDIInfo object that contains all of the file-level and variable-level 
+metadata for the IPUMs extract.
+
+# Examples
+
+Let's assume we have an extract DDI file named `my_extract.xml`
+```julia-repl
+julia> typeof(parse_ddi("my_extract.xml"))
+IPUMS.DDIInfo
+```
+"""
+function parse_ddi(filepath::String)
+
+    # check to make sure the provided file is an xml file.
+    splitext(filepath)[2] != ".xml" && throw(ArgumentError("The DDI file: $filepath should be an XML file."))
+    
+    # check to make sure file exists
+    !isfile(filepath) && throw(ArgumentError("The specified file: $filepath does not exist."))
+    
+    # QUESTION: check if file has multiple files -- then have to handle that
+    
+    # read xml file and parse extract level metadata
+    
+    ddifile = EzXML.readxml(filepath)
+    ns = EzXML.namespace(ddifile.root)
+    
+    ddi = DDIInfo(filepath=filepath, _xml_doc=ddifile, _ns=ns)
+
+    _read_ddi_and_parse_extract_level_metadata!(ddi)
+    
+    # get variables metadata from file
+    _get_var_metadata_from_ddi!(ddi)
+    
+    # TODO determine final output format. For now I return the object.    
```

Comment:
Are you planning to address this in another PR?

Reply:
This is resolved now. I have create a data summary as well as saving the variable metadata in the object. I wrote this TODO because I was not totally sure how to setup the connection between the parsing and the import of the data. But now I have a better sense of this.